services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/server/grpc/GoalStateProvisionerServer.java

```diff
@@ -73,14 +88,26 @@
     private GoalStateClient grpcGoalStateClient;
 
     public GoalStateProvisionerServer() {
+        System.setProperty("JAEGER_SERVICE_NAME","alcor-ncm");
```

Comment:
"alcor-ncm" should be in the config file so that we could check in a centralized place.

Reply:
Oh later I found that that it is actually set by opentracing.jaeger.service-name in the application.properties. Why not use the configuration here?

Reply:
This one is actually tricky, we actually have defined that in the `application.properties`, however, when the code is trying to retrieve this value, it always gets an empty String or just null. That's why I used `System.setProperty` to set this value.

Reply:
This could be the spring bean wiring issue. Maybe create an issue to track it, or use //TODO. It is easy to forget :-)

Reply:
Got it, let me add a //TODO in this code then.